### PR TITLE
Close #10. Add support for Unibeautify config file

### LINKED ISSUE #10: Add Unibeautify Configuration file loader support
See https://unibeautify.com/docs/config-file.html 

TODO:
- [x] Add [cosmiconfig](https://github.com/davidtheclark/cosmiconfig)
- [x] Configure `cosmiconfig`:

```typescript
const configModuleName = 'unibeautify';
const configOptions = {
  rcExtensions: true
};
const configExplorer = cosmiconfig(configModuleName, configOptions);
```

- [x] Load configuration with `configExplorer`:

```typescript
const defaultConfig = {};
return this.configExplorer
      .load(this.repoPath)
      .then(result => (result ? result.config : defaultConfig));
```

- [x] Change `beautifierOptions`:

https://github.com/Unibeautify/vscode/blob/master/src/EditProvider.ts#L40-L50

```typescript
const beautifyOptions: LanguageOptionValues = {}; // REPLACE WITH CONFIG FROM COSMICONFIG
const beautifyData: BeautifyData = {
      languageName,
      fileExtension,
      filePath,
      projectPath,
      options: beautifyOptions,
      text,
};
```

### PR DESCRIPTION
**Pull request by Glavin001**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,13 +85,14 @@
         "mocha": "^5.0.4",
         "rimraf": "^2.5.4",
         "typescript": "^2.4.2",
-        "vscode": "^1.1.4"
+        "vscode": "^1.1.14"
```
Comment (szeck87) [inline package.json]: Why 4 to 14?

File: src/EditProvider.ts [hunk 2]
```diff
@@ -82,4 +99,15 @@ export class EditProvider
...
+
+  protected beautifyOptions(filePath: string): Promise<LanguageOptionValues> {
+    const explorerOptions: Cosmiconfig.Options = {
+      rcExtensions: true,
+    };
+    const explorer = cosmiconfig("unibeautify", explorerOptions);
+    const defaultConfig: LanguageOptionValues = {};
+    return explorer
+      .load(filePath)
+      .then(result => (result ? result.config : defaultConfig));
```
Comment (szeck87) [inline src/EditProvider.ts]: I would catch errors just in case

File: package.json (hunk 1 above)
Comment (Glavin001) [reply package.json]: Just updating to latest.

File: src/EditProvider.ts (hunk 2 above)
Comment (Glavin001) [reply src/EditProvider.ts]: Shouldn't be needed but I'll do anyway.

Comment (Glavin001) [reply src/EditProvider.ts]: Done

File: package.json (hunk 1 above)
Comment (szeck87) [reply package.json]: Sorry reading on my phone. I thought this was the unibeautify vscode package itself.

**Comment by Glavin001** [comment]
Published to v0.3.0!

**Pull request merged**